crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -1020,16 +1020,18 @@ isAllocatedAlignedPointer sym w alignment mutability ptr size mem =
 --
 --   The pointer to read from must be concrete and nonnull.  The contents
 --   of the string may be symbolic; HOWEVER, this function will not terminate
---   until it eventually reaches a concete null-terminator or a load error.
+--   until it eventually reaches a concete null-terminator or a load error,
+--   unless a maximum bound is provided via the @gas@ parameter.
 strLen :: forall sym wptr.
   (IsSymInterface sym, HasPtrWidth wptr, Partial.HasLLVMAnn sym) =>
   sym ->
+  Maybe Word {-^ bound on loop/string length -} ->
```

Comment:
We may want to consider making this non-optional (if rather high, by default). IMO it would be best if Crux and SAW never entered infinite loops without at least printing a warning.